examples/documentation_examples/examples.go

```diff
@@ -2956,3 +2957,188 @@ func StableAPIExamples() {
+	client.Database("pets").Drop(ctx)
+	client.Database("retail").Drop(ctx)
+}
+
+func snapshotQueryPetExample(ctx context.Context, t *testing.T, client *mongo.Client) error {
+	// Start Snapshot Query Example 1
+	// Connect to the pets database
+	db := client.Database("pets")
+
+	// Start a session with snapshotting
+	sess, err := client.StartSession(new(options.SessionOptions).SetSnapshot(true))
+	if err != nil {
+		return err
+	}
+	defer sess.EndSession(ctx)
+
+	var adoptablePetsCount int32
+	err = mongo.WithSession(ctx, sess, func(sc mongo.SessionContext) error {
+		// count the adoptable cats
+		adoptableCatsOuput := "adoptableCatsCount"
+		cursor, err := db.Collection("cats").Aggregate(sc, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableCatsOuput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(sc)
+		resp, err := cursor.Current.LookupErr(adoptableCatsOuput)
+		if err != nil {
+			return err
+		}
+		adoptableCatsCount, _ := resp.Int32OK()
+		adoptablePetsCount += adoptableCatsCount
+
+		// count the adoptable dogs
+		adoptableDogsOuput := "adoptableDogsCount"
+		cursor, err = db.Collection("dogs").Aggregate(sc, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableDogsOuput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(sc)
+		resp, err = cursor.Current.LookupErr(adoptableDogsOuput)
+		if err != nil {
+			return err
+		}
+		adoptableDogsCount, _ := resp.Int32OK()
+		adoptablePetsCount += adoptableDogsCount
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	// End Snapshot Query Example 1
+	require.Equal(t, int32(3), adoptablePetsCount, "Expected adoptablePetsCount to be 3, got %v", adoptablePetsCount)
+	return nil
+}
+
+func snapshotQueryRetailExample(ctx context.Context, t *testing.T, client *mongo.Client) error {
+	// Start Snapshot Query Example 2
+	// Connect to the retail database
+	db := client.Database("retail")
+
+	// Start a session with snapshotting
+	sess, err := client.StartSession(new(options.SessionOptions).SetSnapshot(true))
+	if err != nil {
+		return err
+	}
+	defer sess.EndSession(ctx)
+
+	var totalDailySales int32
+	err = mongo.WithSession(ctx, sess, func(sc mongo.SessionContext) error {
+		// count the total daily sales
+		totalDailySalesOutput := "totalDailySales"
+		cursor, err := db.Collection("sales").Aggregate(sc, mongo.Pipeline{
+			bson.D{{"$match",
+				bson.D{{"$expr",
+					bson.D{{"$gt",
+						bson.A{"$saleDate",
+							bson.D{{"$dateSubtract",
+								bson.D{
+									{"startDate", "$$NOW"},
+									{"unit", "day"},
+									{"amount", 1},
+								},
+							}},
+						},
+					}},
+				}},
+			}},
+			bson.D{{"$count", totalDailySalesOutput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(sc)
+		resp, err := cursor.Current.LookupErr(totalDailySalesOutput)
+		if err != nil {
+			return err
+		}
+		totalDailySales, _ = resp.Int32OK()
```

Comment:
Same as the comment above: This can be simplified some and we should handle the `ok` value.

E.g.
```go
totalDailySales, ok := cursor.Current.Lookup(totalDailySalesOutput).Int32OK()
if !ok {
	return fmt.Errorf("failed to find int32 field %q in document %v", totalDailySalesOutput, cursor.Current)
}
```